### PR TITLE
feat: Create igw without public subnets if database subnet igw route is enabled

### PR DESCRIPTION
**Pull request by wolffberg**
## Description
The module supports creating a IGW route for publicly accessible databases in database subnets however the IGW is not created without any public subnets even though they might not be needed for a database-only VPC.

This PR allows the IGW to be created if database subnets exist and `create_database_internet_gateway_route = true`.

## Motivation and Context
VPCs without public subnets cannot have public databases in database subnets. If you are creating a database-only VPC and want to utilize the `database_subnets` variable, this PR allows you to create both public and private databases.

## Breaking Changes
No.

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
__Example not required as this will only fix the expected behavior__
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects
__This has been tested in a working Terraform setup and validated using terraform console as shown below__
```shell
# local.create_vpc && var.create_igw && length(var.public_subnets) > 0 || (length(var.database_subnets) > 0 && var.create_database_internet_gateway_route) ? 1 : 0

# 0 public subnets, 0 database subnets and database igw route enabled
> true && true && length([]) > 0 || (length([]) > 0 && true) ? 1 : 0
0

# 1 public subnet, 0 database subnet and database igw route disabled
> true && true && length([1]) > 0 || (length([]) > 0 && true) ? 1 : 0
1

# 0 public subnets, 1 database subnet and database igw route enabled
> true && true && length([]) > 0 || (length([1]) > 0 && true) ? 1 : 0
1

# 0 public subnets, 1 database subnet and database igw route disabled
> true && true && length([]) > 0 || (length([1]) > 0 && false) ? 1 : 0
0

# 1 public subnet, 1 database subnet and database igw route disabled
> true && true && length([1]) > 0 || (length([1]) > 0 && false) ? 1 : 0
1
```
__If I'm missing any cases or missed anything else please feel free to add them__

- [X] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by wolffberg** [comment]
@antonbabenko Any way we can get this into `master` 🙏

File: main.tf [hunk 1]
```diff
@@ -125,7 +125,7 @@ resource "aws_vpc_dhcp_options_association" "this" {
 ################################################################################
 
 resource "aws_internet_gateway" "this" {
-  count = local.create_vpc && var.create_igw && length(var.public_subnets) > 0 ? 1 : 0
+  count = local.create_vpc && var.create_igw && length(var.public_subnets) > 0 || (length(var.database_subnets) > 0 && var.create_database_internet_gateway_route) ? 1 : 0
```
Comment (bryantbiggs) [inline main.tf:128]:
```suggestion
  count = local.create_vpc && length(concat(var.public_subnets, database_subnets)) > 0 && (var.create_igw || var.create_database_internet_gateway_route) ? 1 : 0
```

Comment (wolffberg) [reply main.tf:128]: This also works

```shell
# 0 public subnets, 0 database subnets and database igw route enabled
true && length(concat([], [])) > 0 && (true || true) ? 1 : 0
0

# 1 public subnet, 0 database subnet and database igw route disabled
true && length(concat([1], [])) > 0 && (true || true) ? 1 : 0
1

# 0 public subnets, 1 database subnet and database igw route enabled
true && length(concat([], [1])) > 0 && (true || true) ? 1 : 0
1

# 0 public subnets, 1 database subnet and database igw route disabled
true && length(concat([], [])) > 0 && (true || false) ? 1 : 0
0

# 1 public subnet, 1 database subnet and database igw route disabled
true && length(concat([1], [1])) > 0 && (true || false) ? 1 : 0
1
```

**Comment by wolffberg** [comment]
@bryantbiggs Can we go ahead with the merge? 🚀

**Comment by wolffberg** [comment]
Not stale

**Comment by wolffberg** [comment]
Not stale

**Comment by wolffberg** [comment]
@antonbabenko @bryantbiggs 
Hi guys,
I would love if we could do something about this PR.
We should either close it and think about removing the ability to route traffic from the `database` subnets to an IGW or merge it to allow creation of the IGW even if no public subnets are available. It makes little sense having to waste a `/28` subnet just to have an IGW.

**Comment by wolffberg** [comment]
Not stale

**Comment by wolffberg** [comment]
Not stale

**Comment by wolffberg** [comment]
@bryantbiggs how should we move forward with this?

**Pull request closed without merging**